src/main/java/fr/insalyon/citi/golo/compiler/ParseTreeToGoloIrVisitor.java

```diff
@@ -35,6 +36,10 @@
 
   private GoloCompilationException.Builder exceptionBuilder;
 
+  private static String syntheticName(String base) {
+    return "__$$_" + base + "_" + System.currentTimeMillis();
```

Comment:
Indeed. I actually only extracted existing code. BTW, a more generic implementation will be introduced by the macro branch (mainly for hygienic aspect but reusable in syntactic transformations like these)